### ISSUE TITLE
Installing templates to custom names is not actually supported

### ISSUE THREAD

**Issue by tomjakubowski** [Oblong/obi#147]
TLDR: `obi new mytemplate myproject` looks for `$TEMPLATE_HOME/mytemplate/mytemplate.py`, and not `$TEMPLATE_HOME/mytemplate/$SOMETHING_STANDARD.py`.

Our help says this:

```
obi template install <giturl> [<name>] [--template_home=<path>]
```

This suggests you can provide whatever name you want when you install a template. But, in fact, obi is fussy, and that name has to be some particular value for each template, or `obi new mytemplate myproject` won't work. For example, one user reasonably tried to install a `obi-mosaic-gspeak` template with the short name `mosaic`, but obi failed to find the template afterwards:

```
$ obi new mosaic mezzy-moz
Could not find template mosaic
Expected to find /Users/mattieruth/.local/share/oblong/obi/mosaic/mosaic.py
Installed templates:
mosaic
```

Because the template's "main file" was actually `share/obi/mosaic/mosaic-gspeak.py`.

**Comment by mattieruth** [comment]
note that in this case, the python file `mosaic-gspeak.py` also doesn't match the repo name `obi-mosaic-gspeak` so even if the template had been installed without the optional `[<name>]` field, a call to `obi new` would fail.

**Comment by tomjakubowski** [comment]
yeah, it's like an optional argument that's not actually optional, and in fact must be precisely a certain value that differs per template :mike_thumbs_up:
